code/backend/batch/utilities/integrated_vectorization/AzureSearchIndex.py

```diff
@@ -0,0 +1,176 @@
+            SearchField(
+                name="chunk_id",
+                type=SearchFieldDataType.String,
+                key=True,
+                sortable=True,
+                filterable=True,
+                facetable=True,
+                analyzer_name="keyword",
+            ),
+        ]
+
+        vector_search = self.get_vector_search_config()
+
+        semantic_search = self.get_semantic_search_config()
+
+        index = SearchIndex(
+            name=self.env_helper.AZURE_SEARCH_INDEX,
+            fields=fields,
+            vector_search=vector_search,
+            semantic_search=semantic_search,
+        )
+        result = self.index_client.create_or_update_index(index)
+        logger.info(f"{result.name} index created successfully.")
+        return result
+
+    def get_vector_search_config(self):
+        if self.env_helper.is_auth_type_keys():
+            azure_open_ai_parameters = AzureOpenAIParameters(
+                resource_uri=self.env_helper.AZURE_OPENAI_ENDPOINT,
+                deployment_id=self.env_helper.AZURE_OPENAI_EMBEDDING_MODEL,
+                api_key=self.env_helper.OPENAI_API_KEY,
+            )
+        else:
+            azure_open_ai_parameters = AzureOpenAIParameters(
+                resource_uri=self.env_helper.AZURE_OPENAI_ENDPOINT,
+                deployment_id=self.env_helper.AZURE_OPENAI_EMBEDDING_MODEL,
+            )
+
+        return VectorSearch(
+            algorithms=[
+                HnswAlgorithmConfiguration(
+                    name="myHnsw",
+                    parameters=HnswParameters(
+                        m=4,
+                        ef_construction=400,
+                        ef_search=500,
+                        metric=VectorSearchAlgorithmMetric.COSINE,
+                    ),
+                ),
+                ExhaustiveKnnAlgorithmConfiguration(
+                    name="myExhaustiveKnn",
+                    parameters=ExhaustiveKnnParameters(
+                        metric=VectorSearchAlgorithmMetric.COSINE,
+                    ),
+                ),
+            ],
+            profiles=[
+                VectorSearchProfile(
+                    name="myHnswProfile",
+                    algorithm_configuration_name="myHnsw",
+                    vectorizer="myOpenAI",
+                ),
+                VectorSearchProfile(
+                    name="myExhaustiveKnnProfile",
+                    algorithm_configuration_name="myExhaustiveKnn",
+                    vectorizer="myOpenAI",
+                ),
+            ],
+            vectorizers=[
+                AzureOpenAIVectorizer(
+                    name="myOpenAI",
+                    kind="azureOpenAI",
+                    azure_open_ai_parameters=azure_open_ai_parameters,
+                ),
+            ],
+        )
+
+    def get_semantic_search_config(self):
+        semantic_config = SemanticConfiguration(
+            name="my-semantic-config",
```

Comment:
We could use `AZURE_SEARCH_SEMANTIC_SEARCH_CONFIG` from the env helper for the name